packages/nimble-components/src/wafer-map/modules/event-handler.ts

```diff
@@ -0,0 +1,64 @@
+import type { WaferMap } from '..';
+import type { Dimensions, WaferMapQuadrant } from '../types';
+import type { DataManager } from './data-manager';
+import { HoverHandler } from './hover-handler';
+import type { RenderingModule } from './rendering';
+import { ZoomHandler } from './zoom-handler';
+
+/**
+ * EventHandler deals with user interactions and events
+ */
+export class EventHandler {
+    private readonly zoomHandler: ZoomHandler;
+    private readonly hoverHandler: HoverHandler;
+
+    public constructor(
+        private readonly canvas: HTMLCanvasElement,
+        private readonly zoomContainer: HTMLElement,
+        private readonly containerDimensions: Dimensions,
+        private readonly dataManager: DataManager,
+        private readonly canvasLength: number,
+        private readonly renderer: RenderingModule,
+        private readonly rect: HTMLElement,
+        private readonly quadrant: WaferMapQuadrant
+    ) {
+        this.zoomHandler = new ZoomHandler(
+            canvas,
+            zoomContainer,
+            containerDimensions,
+            canvasLength
+        );
+
+        this.hoverHandler = new HoverHandler(
+            canvas,
+            rect,
+            this.zoomHandler,
+            dataManager,
+            quadrant
+        );
+        this.hoverHandler.createHoverDie();
+
```

Comment:
Some initial thoughts in response to @munteannatan's offline question to me about architectures for reducing the coupling between zoom handling and hover handling.

I like the idea of having separate classes for zoom and hover and then one more class (this one) to coordinate between them. But I agree the code seems too tightly coupled right now.

Some ideas that might help (initial thoughts, I can take a deeper look tomorrow):
1. hover handler should not know about zoom handler. The parent handler should provide it with what it needs from the zoom handler. It looks like this is mostly just the zoom transform. A couple cleaner approaches to consider:
   - the hover handler could have a `zoomTransform` property which the parent handler sets whenever the zoom transform changes (I like this approach more if it's possible). 
   - you could create an interface like `ZoomTransformable { getZoomTransform(): ZoomTransform; }` which the zoom handler implements and the hover handler accesses.
2. hover and zoom handler classes should have a smaller API surface area that encapsulates more of what they do.
    - for hover you could make the `mousemove` and `mouseout` functions private and have it attach its own event handlers (would require passing the `WaferMap` to the hover handler).
    - for hover you could also try replacing `toggleHoverDie()` and `createHoverDie()` with a simple property like `hoverDieVisible`. When the component is constructed or the `hoverDieVisible` property is set to true it would do what `createHoverDie` currently does. And when `hoverDieVisible` is set to `false` it would do what `toggleHoverDie(false)` currently does
    - I didn't have a chance to look closely at the API of the zoom handler closely, but hopefully it could be simplified to something that fires events when the zoom is starting/ending and has a way to get the current transform 
3. we should use a more first class mechanism to fire events from handlers. [This article](https://hackwild.com/article/event-handling-techniques/) shows two options. 
    - This code is kind of using the "Event Property Handlers" approach, but it might be more readable if we renamed the callbacks with a more "event-like" naming scheme. I can't tell exactly what the callbacks do but maybe `renderingFunction` is `on-zoom-transform-changed`/`on-zoom-ending` and `hideHoverDieFunction` is `on-zoom-starting`? The names should be from the zoom handler's perspective (what happened to cause the event to fire) not from the parent handler's perspective (what it's doing in response to the event)
    - We could also try the "Event Listeners with EventTarget" approach to make it more obvious that these are events and to allow for additional subscribers in the future.